driver-sync/src/main/com/mongodb/client/internal/ClientSessionBinding.java

```diff
@@ -32,21 +38,46 @@
  * This class is not part of the public API and may be removed or changed at any time.
  */
 public class ClientSessionBinding implements ReadWriteBinding {
-    private final ReadWriteBinding wrapped;
+    private ReadWriteBinding wrapped;
     private final ClientSession session;
     private final boolean ownsSession;
     private final ClientSessionContext sessionContext;
+    private final Cluster cluster;
+    private ReadPreference readPreference;
 
     public ClientSessionBinding(final ClientSession session, final boolean ownsSession, final ReadWriteBinding wrapped) {
```

Comment:
I can't do this since `wrapped` can be either a `ClusterBinding` or a `SingleServerBinding`. I can add a `getCluster` method to `SingleServerBinding` so that the `cluster` field can be removed.

Reply:
I was able to remove the `readPreference` field.

Reply:
You should still change the parameter type to `ClusterBinding` though.  It invites confusion to take a `ReadWriteBinding` and then have an unconditional cast that will fail if the actual type is not `ClusterBinding`.

Reply:
Done